ddtrace/ext/sql.py

```diff
@@ -44,32 +44,37 @@ def _dd_parse_pg_dsn(dsn):
 # Do not import from psycopg directly! This reference will be updated at runtime to use
 # a better implementation that is provided by the psycopg library.
 # This is done to avoid circular imports.
-parse_pg_dsn = _dd_parse_pg_dsn
+
+# Parser for psycopg2
+parse_pg_dsn2 = _dd_parse_pg_dsn
```

Comment:
Perhaps we can have a single global `_parse_pg_func` that we set via the import hook below, and then export a single helper, e.g. `parse_pg_dsn` that encapsulates the common logic
```python
        def parse_pg_dsn(conn):
            global _parse_pg_func

            if hasattr(conn, "dsn"):
                return _parse_pg_func(conn.dsn)
            elif hasattr(conn, "info") and hasattr(conn.info, "dsn"):
                return _parse_pg_func(conn.info.dsn)
            return None
```
?

Reply:
Sure, although I think we still need separate `_parse_pg_func`s, one for each psycopg version? In any case, changed it to pull the logic around getting the DSN from the connection into a shared utility function. Thanks!